software/due-date-changes.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import re
+import sys
+import time
+
+from datetime import datetime
+from jira import JIRA
+
+
+def dump(obj):
+    for attr in dir(obj):
+        print("obj.%s = %r" % (attr, getattr(obj, attr)))
+
+
+def print_help(stream=sys.stderr):
+    help_msg = """Usage: {0} [--projects] LIST,OF,PROJECTS
+"""
+    stream.write(help_msg.format(sys.argv[0]))
+
+
+def parse_args():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-projects", 
+        help="Comma-separated list of project names (ie. HTCONDOR,SOFTWARE)",
+        default="HTCONDOR,SOFTWARE")
+    parser.add_argument("-o", 
+        help="Output file",
+        default=None)
+    args = parser.parse_args()
+
+    projects = args.projects
+    output_file = args.o
+
+    return {
+        "projects": projects,
+        "output_file": output_file
+    }
+
+
+def main():
+
+    # Parse arguments. Assume any error handling happens in parse_args()
+    try:
+        args = parse_args()
+    except Exception as err:
+        print(f"Failed to parse arguments: {err}", file=sys.stderr)
+
+    projects = args["projects"]
+    output_file = args["output_file"]
+
+    # Connect to Jira
+    options = {"server": "https://opensciencegrid.atlassian.net"}
+    jira = JIRA(options)
+
+    # Output CSV headers
+    output_data = "Issue key,Assignee,Original Due Date,Current Due Date,Number of Due Date Changes\n"
+
+    # Now iterate over the projects and list their issues
+    for project in projects.split(","):
+        issues = jira.search_issues(f"project = {project} AND status != Backlog", expand="changelog", maxResults=False)
```

Comment:
So, i think the HTCONDOR project has a "status" called "Backlog", but not every project does.

The SOFTWARE project, for instance, calls this status "Open".

There is also a thing called "statusCategory", which is more general, and as far as i am aware applies to all projects.  This has possible values "To Do", "In Progress", and "Done".  So, i think what you really want here is to refer to "statusCategory" "To Do" rather than "status" "Backlog".

Something like

```suggestion
        issues = jira.search_issues(f"project = {project} AND statusCategory != \"To Do\"", expand="changelog", maxResults=False)
```

But then, why are you excluding Backlog issues, in particular?

Do you want to include Done issues but not Backlog?  (as seems to be the case)

Reply:
Thanks, and good catch on the statusCategory fix! I thought "Backlog" was a built-in Jira state.

As for excluding Backlog issues but not Done, this was by request of the PATh management team. I expect the requirements will change once they start seeing real data.